### PR TITLE
snap,snappy,store: prep to support summary, officialise the side metadata as SideInfo embedded in Info

### PR DESCRIPTION
**Pull request by pedronis**
This tries to accomplish two things:

* prepare us to support the 'summary' field from the store (parallel to what we do with the 'description')
* tries to officialise the concept and list of fields of metadata that either don't come from snap.yaml or for which the store is the canonical source, this is done through SideInfo embedded in Info and splitting for example name into Info.SuggestedName vs SideInfo.OfficialName and having an accessor on Info Name that picks the best available

### CODE REVIEW

**Comment by pedronis** [comment]
retest this please

**Comment by niemeyer** [comment]
LGTM!

**Comment by elopio** [comment]
retest this please

**Comment by elopio** [comment]
retest this please

File: snappy/common_test.go [hunk 1]
```diff
@@ -106,25 +106,26 @@ apps:
...
 	}
-	info := &snap.Info{
-		Name:        name,
-		Version:     version,
-		Developer:   developer,
-		Description: desc,
-		Channel:     channel,
+	sideInfo := snap.SideInfo{
+		OfficialName: name,
+		//Version:           version,
```
Comment (mvo5) [inline snappy/common_test.go]: Would it make sense to remove this commented out line? It seems like we do not really need `Version` in SideInfo (revision yes).

File: snappy/snap_local.go [hunk 2]
```diff
@@ -20,6 +20,8 @@
 package snappy
 
 import (
+	//"fmt"
```
Comment (mvo5) [inline snappy/snap_local.go]: Would it make sense to remove this?

**Comment by mvo5** [comment]
Looks good, thanks for this branch! Two nitpicky comments but no blockers.

**Pull request merged**